File: backend/app/schemas/review.py
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, Field


class ReviewQueueItem(BaseModel):
    id: int
    workflow_type: str
    thread_id: str
    status: str
    document_id: Optional[int] = None
    document_name: Optional[str] = None
    created_at: datetime = Field(
        description="When the workflow run was started.",
    )


class ReviewQueueResponse(BaseModel):
    pending_approvals: List[ReviewQueueItem]
    total: int

File: frontend/src/types/api/review.ts
import type { WorkflowExecutionStatus } from "@/types/api/workflow-execution";

export interface ReviewItem {
  id: number;
  workflow_type: string;
  thread_id: string;
  status: WorkflowExecutionStatus | string;
  document_id: number | null;
  document_name: string | null;
  created_at: string;
}

export interface ReviewQueueResponse {
  pending_approvals: ReviewItem[];
  total: number;
}

export interface ListReviewsParams {
  limit?: number;
}
